src/main/java/com/app/onlinebookstore/repository/UserRepository.java

```diff
@@ -0,0 +1,13 @@
+package com.app.onlinebookstore.repository;
+
+import com.app.onlinebookstore.model.User;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface UserRepository extends JpaRepository<User, Long> {
+    @Query("SELECT u FROM User u LEFT JOIN FETCH u.roles WHERE u.email = :email")
```

Comment:
I think your method will work even without this query. Use the power of Spring Data